### PR TITLE
Use ubuntu apt repo if we are on ubuntu

### LINKED ISSUE #277: DEB repo on ubuntu should not point to debian
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 6.0.2
- Ruby:
- Distribution: Ubuntu 18.04
- Module version: 3.0.0

## How to reproduce (e.g Puppet code you use)
```
include gitlab
```

## What are you seeing

```
cat /etc/apt/sources.list.d/gitlab_official_ce.list
# Official repository for GitLab Omnibus
deb https://packages.gitlab.com/gitlab/gitlab-ce/debian bionic main
```

## What behaviour did you expect instead

```
cat /etc/apt/sources.list.d/gitlab_official_ce.list
# Official repository for GitLab Omnibus
deb https://packages.gitlab.com/gitlab/gitlab-ce/ubuntu bionic main
```

## Any additional information you'd like to impart
Seeing this while upgrade the module to 3.0.0

```
Notice: /Stage[main]/Gitlab::Omnibus_package_repository/Apt::Source[gitlab_official_ce]/Apt::Setting[list-gitlab_official_ce]/File[/etc/apt/sources.list.d/gitlab_official_ce.list]/content:
--- /etc/apt/sources.list.d/gitlab_official_ce.list     2018-06-10 23:24:55.000000000 +0200
+++ /tmp/puppet-file20181013-9939-zso5lj        2018-10-13 18:44:21.000000000 +0200
@@ -1,4 +1,3 @@
 # This file is managed by Puppet. DO NOT EDIT.
-# Official repository for Gitlab
-deb https://packages.gitlab.com/gitlab/gitlab-ce/ubuntu/ bionic main
-deb-src https://packages.gitlab.com/gitlab/gitlab-ce/ubuntu/ bionic main
+# Official repository for GitLab Omnibus
+deb https://packages.gitlab.com/gitlab/gitlab-ce/debian bionic main

Notice: /Stage[main]/Gitlab::Omnibus_package_repository/Apt::Source[gitlab_official_ce]/Apt::Setting[list-gitlab_official_ce]/File[/etc/apt/sources.list.d/gitlab_official_ce.list]/content: current_value '{md5}56bb8259cb0c7c33ec4272ddc15ce629', should be '{md5}4381bb154bd6bece2012d14b34445842' (noop)
Info: /Stage[main]/Gitlab::Omnibus_package_repository/Apt::Source[gitlab_official_ce]/Apt::Setting[list-gitlab_official_ce]/File[/etc/apt/sources.list.d/gitlab_official_ce.list]: Scheduling refresh of Class[Apt::Update]
```

It also breaks `apt update`

```
# apt update
Hit:1 http://mirror.hetzner.de/ubuntu/packages bionic InRelease
Hit:2 http://mirror.hetzner.de/ubuntu/security bionic-security InRelease
Hit:3 http://mirror.hetzner.de/ubuntu/packages bionic-updates InRelease
Ign:4 http://apt.puppetlabs.com bionic InRelease
Hit:5 http://apt.puppetlabs.com bionic Release
Hit:7 https://packages.gitlab.com/runner/gitlab-runner/ubuntu bionic InRelease
Ign:8 https://packages.gitlab.com/gitlab/gitlab-ce/debian bionic InRelease
Err:9 https://packages.gitlab.com/gitlab/gitlab-ce/debian bionic Release
  404  Not Found [IP: 2a01:4f8:173:282c:eebf:1cb6:bdc2:e92 3128]
Reading package lists... Done
E: The repository 'https://packages.gitlab.com/gitlab/gitlab-ce/debian bionic Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

### PR DESCRIPTION
**Pull request by jkroepke**
Fixes #277

### CODE REVIEW

**Comment by bastelfreak** [comment]
Thanks @jkroepke !

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -25,6 +25,14 @@
           it { is_expected.to contain_apt__source('gitlab_official_ee').with_ensure('absent') }
           it { is_expected.not_to contain_apt__source('gitlab_official_') }
           it { is_expected.not_to contain_yumrepo('gitlab_official_ce') }
+          case facts[:operatingsystem]
```
Comment (LongLiveCHIEF) [inline spec/classes/init_spec.rb]: @jkroepke if you get a chance can you submit a pull that uses modern fact for this?

**Comment by LongLiveCHIEF** [comment]
Thanks for the PR, much appreciated! Any other issues installing on Ubuntu?

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (jkroepke) [reply spec/classes/init_spec.rb]: Last time I'm writing spec test, modern facts are simply not work or not defined by the module you are using for generated the facts. 

No idea whats the current state is.

**Comment by jkroepke** [comment]
> Any other issues installing on Ubuntu?

I just upgraded from 2.x, puppet changed nothing expect the apt file.

I also missed the CI Runner part right now. It's fine to split runner and gitlab into a separate modules, but you should release the runner module. Otherwise a lot of people can not upgrade to the current 3.0.

At the end thanks for still supporting this over all the years.

File: spec/classes/init_spec.rb (hunk 1 above)
Comment (Dan33l) [reply spec/classes/init_spec.rb]: For information :
structured facts with rspec-puppet : `facts[:os]['family']`
structured facts with beaker-rspec : `fact('os.family')`

**Pull request merged**